WIP: Start of documentation

JonTheNiceGuy opened the pull request:
Please can someone review what I've done thus far and see if it's useful? If so, I'll carry on, on this PR!

rawdigits commented:
This seems like a great start! I think we need things like this to expand on the basic + limited example from the main README.md. As you add to it, we will go over it in technical detail just to be another set of eyes, but I like that you are covering some of the non-obvious features like DNS here.

JonTheNiceGuy commented:
OK, so I've added more content to this file, and it's ready for a tech vet. There are so TODO items, which are outside my scope of knowledge.

I've also added two simple-config-{lighthouse|node}.yml files for a "simple" setup that match what I imagine most people are likely to do with nebula. If you want those two dropping out into a separate PR, I'm more than happy to do so!

rawdigits commented:
Just wanted to let you know I gave this a read through. Thanks much for doing some forensic engineering to discover the features we haven't documented yet.

In the coming week, I'll give feedback on a couple of details that may be slightly incorrect in the doc, but overall this makes for a great start.

(I'll also nag someone else on the core team to review this!) :)

JonTheNiceGuy commented:
So, I've added some updates to this based on how my tickets I've raised have been responded to. If someone with more technical knowledge can cover off some of the TODO items, or at least comment on them here, I can update them accordingly...

JonTheNiceGuy commented:
Updated based on info from #142 on subnets.

JonTheNiceGuy commented:
Added more detail from 142.

JonTheNiceGuy commented:
@rawdigits just a small nudge that you said you'd review this, and arrange for someone else to take a look too. Not a problem, just wanted to ensure this isn't lost in the noise :)

rawdigits commented:
@JonTheNiceGuy the entire team is in the same physical location for the first time in ages and I plan to go over this with them while we are in the same place. 
(👋 @wadey and @nbrownus )

auto-mation commented on Documentation/README.md:
```diff
@@ -0,0 +1,745 @@
+machines. Those machines must not be able to talk to the network, and so should
+be added as a list of "fingerprints" to ignore.
+
+<!-- TODO: Document how to get the fingerprints -->
```
To retrieve the thumbprint, simply use 

`./nebula-cert print -path /path/to/your/cert.crt | grep Fingerprint | awk {'print $2'}`

JonTheNiceGuy replied:
Added (practically) verbatim in 8bc14b0

toonsevrin commented:
@JonTheNiceGuy, can you move the documentation to `/docs`? It's convention and allows for more straightforward github-pages integration.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/slackhq/nebula?pullRequest=119) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/slackhq/nebula?pullRequest=119) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/slackhq/nebula?pullRequest=119) it.</sub>

JonTheNiceGuy commented:
Just to let people know, I've stopped looking at this, as I was being drawn in too many directions. If you want to merge it - please feel free ✔ If you don't, there are no worries and no hard feelings 👍 but if you need changes to be made, please take this, merge it and fix it as I don't have the time or space or capacity at the moment to make anything with this ❤

kotp commented on Documentation/README.md:
```diff
@@ -0,0 +1,750 @@
+
+[In the `tun` block, we only *need* to set one value, and that's the device name 
+(or `dev`). While it's possible to tweak these settings, the "best" defaults
+are already set. You can have a peak at your options in the 
```
```suggestion
are already set. You can have a peek at your options in the 
```

kotp commented on Documentation/simple-config-lighthouse.yml:
```diff
@@ -0,0 +1,48 @@
+      host: any
+    - port: 53
+      proto: any
+      host: any
```
Missing EOL

nbrownus commented:
@JonTheNiceGuy thanks again for all the work putting this together. We have decided to host all documentation at https://www.defined.net/nebula/ which is backed by this open source repo https://github.com/DefinedNet/nebula-docs.
